src/data/chats/mod.rs

```diff
@@ -194,26 +198,29 @@ impl Chats {
 
     pub fn create_empty_chat_and_load_file(&mut self, file: &File) {
         let mut new_chat = Chat::new(self.chats_dir.clone());
+        let id = new_chat.id;
         new_chat.associated_entity = Some(ChatEntityId::ModelFile(file.id.clone()));
         new_chat.save();
```

Comment:
I'm concerned about file writes happening more than necessary here as writes to disk are a precious resource.

I can remove this and other `save` statements as they are performed by `set_current_chat` internally. But would not be clear here that the chat is being saved (without a comment), and there is the risk of removing the `save` on `set_current_chat` affecting this other places).

Another approach would be to buffer writes, so if you do many `.save()` in a second, only the last one is flushed. I tried adding a function that does exactly this (see `buffered_write_to_file` in `filesystem.rs`) in the simplest and smallest way I could think about. But doing that properly would require async code, separate tasks per file, non-blocking error handling, etc.

Please let me know if you don't agree on adding that function and I will remove it, no question asked.